freeipa/src/main/java/com/sequenceiq/freeipa/sync/ProviderChecker.java

```diff
@@ -77,6 +79,28 @@
         }, LOGGER, ":::Auto sync::: provider is checked in {}ms");
     }
 
+    public Set<InstanceMetaData> updateStatuses(Stack stack, InstanceStatus instanceStatus, boolean updateStatusFromFlow) {
+        Set<InstanceMetaData> checkableInstances = stack.getAllInstanceMetaDataList().stream()
+                .filter(Predicate.not(InstanceMetaData::isTerminated))
+                .filter(Predicate.not(InstanceMetaData::isDeletedOnProvider))
+                .collect(Collectors.toSet());
+
+        if (!updateStatusFromFlow && flowLogService.isOtherFlowRunning(stack.getId())) {
```

Comment:
I don't think this is necessary as it is only called from one flow step

Reply:
Yes. You are right. I removed it.